descartes_light/test/src/perf_plotter.py

```diff
@@ -0,0 +1,175 @@
+            self.capped_interp(self.dict[alg]['n_waypoints'][x], 150, 1),
+            self.capped_interp(self.dict[alg]['Graph Search Time'][x], 2, 1))
+            for x in range(0, num_tests) ]
+
+        y_vals = [self.dict[alg]['n_waypoints'][x]+(.25*x)  for x in range(0, num_tests)  ]
+        
+        # Generate plot, amazingly just a single command thanks to matplotlib:
+        ax.bar(self.dict[alg]['samples'], self.dict[alg]['Graph Search Time'], y_vals, 
+             zdir='y', width=12, color=test_color_tuples, alpha=0.75 )
+
+        a = list(set(tuple(zip(self.dict[alg]["samples"], self.dict[alg]["n_waypoints"]))))
+        b = zip(self.dict[alg]["samples"], self.dict[alg]["n_waypoints"], self.dict[alg]["Graph Search Time"])
+        zs = [0]*len(a)
+        count = [0]*len(a)
+        #Print a time in seconds for each cluster of data
+        for x, y, z in b:
+            if (x,y) in a:
+                n = a.index((x,y))
+                # if(n==3):
+                #     print('Index of cluster ' + str(x) + " " + str(y) + " is " + str(n) + " and we're adding " + str(z) + "+"+ str(zs[n]) +"=") 
+                zs[n] = (zs[n] + z)
+                # if(n==3):
+                #     print("Running sum: " + str(zs[n]))
+                count[n] += 1.0
+            else:
+                print("Index Error! Can't find cluster!")
+        avg_z = [zs[i]/count[i] for i in range(len(zs))]
+
+        b = zip(a, avg_z)
+        for (x, y), z in b:
+            if (x,y) in a:
+                #If the xy cluster hasn't been labeled yet:
+                ax.text(x, y, z+.25, str(np.round(z, 3)), color=(0,0,.8), fontsize=8, zdir='y')
+                #del a[a.index(x,y)]
+            else:
+                print("Index Error! Can't find cluster!")
+        # ax.text(10, 10, -.5, "Blah", zdir='y')
+
+        # Format graph
+        ax.set_xlabel('Samples')
+        ax.set_ylabel('# of Waypoints')
+        ax.set_zlabel('Search Time (s)')
+        # On the axes let's only label the discrete values that we have data for.
+        ax.set_yticks(self.dict[alg]['n_waypoints'])
+        ax.set_xticks(self.dict[alg]['samples'])
+        if(max != None):
+            ax.set_zlim(top = max+.1)
+        ax.set_title(alg, fontsize=8)
+
+    def capped_interp(self, input, max, cap):
+        return min(input/max, cap)
+
+    def main(self):
+        
+
+        self.strip()
+        solvers_list = list(self.dict)
+        # print("An example algorithm's performance: " + solvers_list[-3])
+        # print(str(self.dict[solvers_list[-3]]))
+        if len(sys.argv) > 1:
+            # argv[1] has your filename
+            input1 = sys.argv[1]
+            print ("Requested single graph of index " + input1)
+            self.plot(solvers_list, 1, int(input1))
+        else:
+            self.plot(solvers_list, len(solvers_list))
+        
+if __name__ == '__main__':
+    application = Perf_Plotter()
+    application.main()
```

Comment:
It's usually not a good idea to declare variables here because they are in the global scope, which means `application` is available to be used by any other function in this script. The better route would be to make `main` a free function (rather than a class member) and to call it here:

```suggestion
def main():
    # Parse command line arguments
    # Create plotter object
    # Parse the input file
    # Plot the benchmarks


if __name__ == '__main__':
    main()
```